lib/puppet/file_serving/s3_metadata.rb

```diff
@@ -0,0 +1,22 @@
+require 'puppet/file_serving/metadata'
+
+class Puppet::FileServing::S3Metadata < Puppet::FileServing::Metadata
+  def initialize(s3_response, path = '/dev/null')
+    super(path)
+
+    # ignore options that do not apply to s3 metadata
+    @owner = @group = @mode = nil
+
+    if s3_response.etag
+      @checksum_type = 'md5'
+      # Strip the double quotes from the S3 etag response
+      @checksum = "{md5}#{s3_response.etag.tr('"','')}"
```

Comment:
In the majority of instances the etag will be the md5 hash of the file's content.

However there are certain instances where the etag will not be the md5 hash.  These instances are when:
1. multipart uploads;
2. files are over a certain size (5GB's?);
3. using customer-provided encryption keys during the upload process.

References:
http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPUT.html
http://stackoverflow.com/questions/6591047/etag-definition-changed-in-amazon-s3
http://stackoverflow.com/questions/1775816/how-to-get-the-md5sum-of-a-file-on-amazons-s3